ui/admin/tests/acceptance/credential-store/read-test.js

```diff
@@ -54,53 +56,68 @@ module('Acceptance | credential-stores | read', function (hooks) {
+  test('visiting static credential store', async function (assert) {
     assert.expect(2);
-    await visit(urls.credentialStores);
+    await click(`[href="${urls.credentialStores}"]`);
     await a11yAudit();
     assert.strictEqual(currentURL(), urls.credentialStores);
-    await visit(urls.staticCredentialStore);
+
+    await click(`[href="${urls.staticCredentialStore}"]`);
     await a11yAudit();
+
     assert.strictEqual(currentURL(), urls.staticCredentialStore);
   });
 
-  test('visiting vault credential store in /credential-stores', async function (assert) {
+  test('visiting vault credential store', async function (assert) {
     assert.expect(2);
-    await visit(urls.credentialStores);
-    await a11yAudit();
+    await click(`[href="${urls.credentialStores}"]`);
     assert.strictEqual(currentURL(), urls.credentialStores);
-    await visit(urls.vaultCredentialStore);
+
+    await click(`[href="${urls.vaultCredentialStore}"]`);
     await a11yAudit();
+
     assert.strictEqual(currentURL(), urls.vaultCredentialStore);
   });
 
   test('cannot navigate to a static credential store form without proper authorization', async function (assert) {
-    assert.expect(1);
+    assert.expect(2);
     instances.staticCredentialStore.authorized_actions =
       instances.staticCredentialStore.authorized_actions.filter(
         (item) => item !== 'read'
       );
-    await visit(urls.staticCredentialStore);
-    assert.notOk(find('main tbody .rose-table-header-cell:nth-child(1) a'));
+
+    await click(`[href="${urls.credentialStores}"]`);
+
+    assert.dom(`[href="${urls.staticCredentialStore}"]`).doesNotExist();
+    assert.dom(`[href="${urls.vaultCredentialStore}"]`).isVisible();
   });
 
   test('cannot navigate to a vault credential store form without proper authorization', async function (assert) {
-    assert.expect(1);
+    assert.expect(2);
     instances.vaultCredentialStore.authorized_actions =
       instances.vaultCredentialStore.authorized_actions.filter(
         (item) => item !== 'read'
       );
-    await visit(urls.vaultCredentialStore);
-    assert.notOk(find('main tbody .rose-table-header-cell:nth-child(1) a'));
+
+    await click(`[href="${urls.credentialStores}"]`);
+
+    assert.dom(`[href="${urls.vaultCredentialStore}"]`).doesNotExist();
+    assert.dom(`[href="${urls.staticCredentialStore}"]`).isVisible();
   });
 
-  test('visiting an unknown credential store display 404 message', async function (assert) {
-    assert.expect(1);
+  test('visiting an unknown credential store displays 404 message', async function (assert) {
+    assert.expect(2);
+    await visit(urls.credentialStores);
+    assert.dom(`[href="${urls.unknownCredentialStore}"]`).doesNotExist();
+
     await visit(urls.unknownCredentialStore);
     await a11yAudit();
+
     assert.strictEqual(
       find('.rose-message-subtitle').textContent.trim(),
       'Error 404'
```

Comment:
While this `strictEqual` works, I think the `.hasText()` is more clear and easier to read.
```suggestion
    assert.dom(find('.rose-message-subtitle'))
      .hasText('Error 404');
```

Reply:
I think the `find` here is unnecessary as `assert.dom` will do the finding.

Reply:
Right. Thanks for pointing that out. Another reason I like `assert.dom`